Index: Makefile.am
===================================================================
RCS file: /cvs/systemtap/src/Makefile.am,v
retrieving revision 1.80
retrieving revision 1.81
diff -u -r1.80 -r1.81
--- Makefile.am	2 Jul 2007 19:40:50 -0000	1.80
+++ Makefile.am	3 Jul 2007 19:49:36 -0000	1.81
@@ -44,16 +44,16 @@
 
 .PHONY: install-elfutils
 install-elfutils:
-	mkdir -p $(pkglibdir)
+	mkdir -p $(DESTDIR)$(pkglibdir)
 	for file in lib-elfutils/*.so* lib-elfutils/${PACKAGE_NAME}/*.so*; do \
-	   $(INSTALL_PROGRAM) $$file $(pkglibdir); \
+	   $(INSTALL_PROGRAM) $$file $(DESTDIR)$(pkglibdir); \
 	done
 install-exec-local: install-elfutils
 endif
 
 staprun_SOURCES = runtime/staprun/staprun.c runtime/staprun/mainloop.c \
 	runtime/staprun/symbols.c runtime/staprun/ctl.c \
-	runtime/staprun/relay.c runtime/staprun/relay_old.c 
+	runtime/staprun/relay.c runtime/staprun/relay_old.c
 
 staprun_CFLAGS = @PROCFLAGS@ $(AM_CFLAGS)
 staprun_LDADD = @PROCFLAGS@ -lpthread
@@ -78,7 +78,7 @@
 	$(CC) -shared -rdynamic $(LDFLAGS) $(CFLAGS) -fPIC -o $@ $<
 all-local: $(STAPLOG)
 install-exec-local: $(STAPLOG)
-	$(INSTALL) $(STAPLOG) $(pkglibdir)
+	$(INSTALL) $(STAPLOG) $(DESTDIR)$(pkglibdir)
 else
 endif
 
@@ -153,10 +153,9 @@
   SUBDIRS += runtime/lket/b2a
 endif
 
-SRCDIR = $(shell cd $(srcdir); pwd)
-
 check:
-	$(MAKE) -C testsuite check SYSTEMTAP_RUNTIME=$(SRCDIR)/runtime SYSTEMTAP_TAPSET=$(SRCDIR)/tapset LD_LIBRARY_PATH=$(PWD)/lib-elfutils:$(PWD)/lib-elfutils/systemtap SYSTEMTAP_PATH=$(PWD) RUNTESTFLAGS="$(RUNTESTFLAGS)"
+	SRCDIR=`cd $(srcdir); pwd`; \
+	$(MAKE) -C testsuite check SYSTEMTAP_RUNTIME=$$SRCDIR/runtime SYSTEMTAP_TAPSET=$$SRCDIR/tapset LD_LIBRARY_PATH=$(PWD)/lib-elfutils:$(PWD)/lib-elfutils/systemtap SYSTEMTAP_PATH=$(PWD) RUNTESTFLAGS="$(RUNTESTFLAGS)"
 
 installcheck:
 	$(MAKE) -C testsuite installcheck RUNTESTFLAGS="$(RUNTESTFLAGS)"
Index: Makefile.in
===================================================================
RCS file: /cvs/systemtap/src/Makefile.in,v
retrieving revision 1.89
retrieving revision 1.90
diff -u -r1.89 -r1.90
--- Makefile.in	2 Jul 2007 19:40:50 -0000	1.89
+++ Makefile.in	3 Jul 2007 19:49:36 -0000	1.90
@@ -283,7 +282,7 @@
 @BUILD_ELFUTILS_TRUE@stap_DEPENDENCIES = lib-elfutils/libdw.so
 staprun_SOURCES = runtime/staprun/staprun.c runtime/staprun/mainloop.c \
 	runtime/staprun/symbols.c runtime/staprun/ctl.c \
-	runtime/staprun/relay.c runtime/staprun/relay_old.c 
+	runtime/staprun/relay.c runtime/staprun/relay_old.c
 
 staprun_CFLAGS = @PROCFLAGS@ $(AM_CFLAGS)
 staprun_LDADD = @PROCFLAGS@ -lpthread
@@ -317,7 +316,6 @@
 TEST_COV_DIR = coverage
 # XXX: leaves behind man pages
 SUBDIRS = testsuite $(am__append_4)
-SRCDIR = $(shell cd $(srcdir); pwd)
 all: $(BUILT_SOURCES) config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
@@ -1385,9 +1383,9 @@
 
 @BUILD_ELFUTILS_TRUE@.PHONY: install-elfutils
 @BUILD_ELFUTILS_TRUE@install-elfutils:
-@BUILD_ELFUTILS_TRUE@	mkdir -p $(pkglibdir)
+@BUILD_ELFUTILS_TRUE@	mkdir -p $(DESTDIR)$(pkglibdir)
 @BUILD_ELFUTILS_TRUE@	for file in lib-elfutils/*.so* lib-elfutils/${PACKAGE_NAME}/*.so*; do \
-@BUILD_ELFUTILS_TRUE@	   $(INSTALL_PROGRAM) $$file $(pkglibdir); \
+@BUILD_ELFUTILS_TRUE@	   $(INSTALL_PROGRAM) $$file $(DESTDIR)$(pkglibdir); \
 @BUILD_ELFUTILS_TRUE@	done
 @BUILD_ELFUTILS_TRUE@install-exec-local: install-elfutils
 
@@ -1395,7 +1393,7 @@
 @BUILD_CRASHMOD_TRUE@	$(CC) -shared -rdynamic $(LDFLAGS) $(CFLAGS) -fPIC -o $@ $<
 @BUILD_CRASHMOD_TRUE@all-local: $(STAPLOG)
 @BUILD_CRASHMOD_TRUE@install-exec-local: $(STAPLOG)
-@BUILD_CRASHMOD_TRUE@	$(INSTALL) $(STAPLOG) $(pkglibdir)
+@BUILD_CRASHMOD_TRUE@	$(INSTALL) $(STAPLOG) $(DESTDIR)$(pkglibdir)
 
 # Copy some of the testsuite sample scripts to the distdir
 # 'examples/samples' directory.
@@ -1443,7 +1441,8 @@
 	-rm -rf $(DESTDIR)$(localstatedir)/cache/$(PACKAGE)
 
 check:
-	$(MAKE) -C testsuite check SYSTEMTAP_RUNTIME=$(SRCDIR)/runtime SYSTEMTAP_TAPSET=$(SRCDIR)/tapset LD_LIBRARY_PATH=$(PWD)/lib-elfutils:$(PWD)/lib-elfutils/systemtap SYSTEMTAP_PATH=$(PWD) RUNTESTFLAGS="$(RUNTESTFLAGS)"
+	SRCDIR=`cd $(srcdir); pwd`; \
+	$(MAKE) -C testsuite check SYSTEMTAP_RUNTIME=$$SRCDIR/runtime SYSTEMTAP_TAPSET=$$SRCDIR/tapset LD_LIBRARY_PATH=$(PWD)/lib-elfutils:$(PWD)/lib-elfutils/systemtap SYSTEMTAP_PATH=$(PWD) RUNTESTFLAGS="$(RUNTESTFLAGS)"
 
 installcheck:
 	$(MAKE) -C testsuite installcheck RUNTESTFLAGS="$(RUNTESTFLAGS)"
